managed-ledger/src/test/java/org/apache/bookkeeper/mledger/impl/ManagedCursorTest.java

```diff
@@ -251,17 +252,36 @@ public void testConcurrentPropertyOperationsThreadSafety() throws Exception {
                             }
                             break;
                     }
+                    completedOperations.incrementAndGet();
+
                 } catch (ConcurrentModificationException cme) {
+                    // Record ConcurrentModificationException but don't fail immediately
```

Comment:
HashMap will throw ConcurrentModificationException only in specific methods such as `forEach`, `compute`, and a few others.

I didn't run the tests without the fix to validate whether ConcurrentModificationException would be thrown. Have you seen it if you revert the fix and run the test?

I think that this catch block can be removed and just fail immediately like it's done for other Exceptions.